### PR TITLE
[bitnami/postgresql] Improvement for network policy

### PR DESCRIPTION
**Pull request by rafariossaa**
**Description of the change**

Improvement for network policy

**Benefits**

**Possible drawbacks**

**Applicable issues**

  - fixes #

**Additional information**

DO NOT MERGE.
This PR is to get the changes reviewed.

**Checklist** 

- [X] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/).
- [X] Variables are documented in the README.md
- [X] Title of the PR starts with chart name (e.g. [bitnami/<name_of_the_chart>])

### CODE REVIEW

File: bitnami/postgresql/values.yaml [hunk 1]
```diff
@@ -726,31 +726,91 @@ resources:
...
-  ## on. When true, PostgreSQL will accept connections from any source
-  ## (with the correct destination port).
-  ##
-  allowExternal: true
-  ## @param networkPolicy.explicitNamespacesSelector A Kubernetes LabelSelector to explicitly select namespaces from which ingress traffic could be allowed
-  ## If explicitNamespacesSelector is missing or set to {}, only client Pods that are in the networkPolicy's namespace
-  ## and that match other criteria, the ones that have the good label, can reach the DB.
-  ## But sometimes, we want the DB to be accessible to clients from other namespaces, in this case, we can use this
-  ## LabelSelector to select these namespaces, note that the networkPolicy's namespace should also be explicitly added.
+  ## @param networkPolicy.metrics.enabled Enable network policy for metrics (prometheus)
```
Comment (pablogalegoc) [inline bitnami/postgresql/values.yaml:732]: Missing `networkPolicy.enabled`

File: bitnami/postgresql/values.yaml [hunk 2]
```diff
@@ -726,31 +726,91 @@ resources:
...
+    ##   label: monitoring
+    ##
+    podSelector: {}
+    ## e.g:
+    ## namespaceSelector:
+    ##   label: monitoring
+    ##
+    namespaceSelector: {}
+  ## @param networkPolicy.ingressRules.primaryAccessOnlyFrom.enabled Enable ingress rule that makes primary mariadb nodes only accessible from a particular origin.
+  ## @param networkPolicy.ingressRules.primaryAccessOnlyFrom.namespaceSelector [object] Namespace selector label that is allowed to access the primary node. This label will be used to identified the allowed namespace(s).
```
Comment (pablogalegoc) [inline bitnami/postgresql/values.yaml:749]: According to Miña, in these cases where we declare empty objects the `[object]` modifier is redundant, so we should remove it ([source](https://github.com/bitnami/charts/pull/8116#discussion_r749182402))

File: bitnami/postgresql/values.yaml [hunk 3]
```diff
@@ -726,31 +726,91 @@ resources:
...
+  ## @param networkPolicy.ingressRules.primaryAccessOnlyFrom.podSelector [object] Pods selector label that is allowed to access the primary node. This label will be used to identified the allowed pod(s).
+  ## @param networkPolicy.ingressRules.primaryAccessOnlyFrom.customRules [object] Custom network policy for the primary node.
+  ## @param networkPolicy.ingressRules.readReplicasAccessOnlyFrom.enabled Enable ingress rule that makes primary mariadb nodes only accessible from a particular origin.
+  ## @param networkPolicy.ingressRules.readReplicasAccessOnlyFrom.namespaceSelector [object] Namespace selector label that is allowed to acces the secondary nodes. This label will be used to identified the allowed namespace(s).
+  ## @param networkPolicy.ingressRules.readReplicasAccessOnlyFrom.podSelector [object] Pods selector label that is allowed to access the secondary nodes. This label will be used to identified the allowed pod(s).
+  ## @param networkPolicy.ingressRules.readReplicasAccessOnlyFrom.customRules [object] Custom network policy for the secondary nodes.
+  ##
+  ingressRules:
+    ## Allow access to the primary node only from the indicated:
```
Comment (pablogalegoc) [inline bitnami/postgresql/values.yaml:758]: Nitpick:
```suggestion
    ## Allow access to the primary node only from the indicated:
    ##
```
Check other cases like this below

**Review by pablogalegoc — COMMENTED** [review]
Thanks @rafariossaa!

I've left some comments below. Also, we will need to update:
- [Network policy](https://github.com/bitnami/charts/tree/master/bitnami/postgresql#networkpolicy) section of the README
- The `NOTES.txt`, since it uses some of the previous parameters ([example](https://github.com/bitnami/charts/blob/master/bitnami/postgresql/templates/NOTES.txt#L56))

**Review by pablogalegoc — APPROVED** [review]

**Review by pablogalegoc — CHANGES_REQUESTED** [review]
Approved but these changes have to be integrated in the next major, so I'm blocking the merge by "requesting changes"

**Comment by juan131** [comment]
These changes have been implemented at https://github.com/bitnami/charts/pull/8119

**Pull request closed without merging**